logic/transaction.js

```diff
@@ -693,9 +690,12 @@ class Transaction {
 		};
 
 		if (checkExists) {
-			this.checkConfirmed(transaction, checkConfirmedErr => {
-				if (checkConfirmedErr) {
-					return setImmediate(cb, checkConfirmedErr);
+			this.checkConfirmed(transaction, (checkConfirmedErr, isConfirmed) => {
+				if (checkConfirmedErr || isConfirmed) {
```

Comment:
We should return different error messages to callback based on whether `checkConfirmredErr` had an error value or `isConfirmed` was set to true.